nixos/modules/services/misc/ntfy-sh.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ntfy-sh;
+
+  settingsFormat = pkgs.formats.yaml { };
+in
+
+{
+  options.services.ntfy-sh = {
+    enable = mkEnableOption "[ntfy-sh](https://ntfy.sh), a push notification service";
+
+    settings = mkOption {
+      type = types.submodule {
+
+        freeformType = settingsFormat.type;
+
+        options.base-url = mkOption {
+          type = types.str;
+          default = "";
+          description = "Public facing base URL of the service.";
+          example = "https://ntfy.sh";
+        };
+      };
+
+      default = { };
```

Comment:
Can the software run with no setting by default?

Reply:
Yes, if no settings are given, the server will simply listen on :80